src/server/game/Entities/Player/Player.cpp

```diff
@@ -4831,7 +4831,7 @@ void Player::RepopAtGraveyard()
 
     bool shouldResurrect = false;
     // Such zones are considered unreachable as a ghost and the player must be automatically revived
-    if ((!IsAlive() && zone && zone->GetFlags().HasFlag(AreaFlags::NoGhostOnRelease)) || GetTransport() || GetPositionZ() < GetMap()->GetMinHeight(GetPhaseShift(), GetPositionX(), GetPositionY()))
+    if ((!IsAlive() && zone && zone->GetFlags().HasFlag(AreaFlags::NoGhostOnRelease)) || GetMap()->IsNonRaidDungeon() || GetMap()->IsRaid() || GetTransport() || GetPositionZ() < GetMap()->GetMinHeight(GetPhaseShift(), GetPositionX(), GetPositionY()))
```

Comment:
I think this should be just `GetMap()->IsDungeon()` (covers dungeon, raid and scenario)

Reply:
There are scenarios that have spirit healers in mists of pandaria, its been too long and im not sure if we can queue for them in dragonflight

Reply:
yes you can queue for them

Reply:
just tested to make sure, yes you get ghost when you die. Maybe not in all of them but I think that might just be done with AreaFlags::NoGhostOnRelease